rpc/api/miner.go

```diff
@@ -126,7 +126,7 @@ func (self *minerApi) SetExtra(req *shared.Request) (interface{}, error) {
 		return nil, err
 	}
 
-	if uint64(len(args.Data)) > params.MaximumExtraDataSize.Uint64()*2 {
+	if uint64(len(args.Data)) > params.MaximumExtraDataSize.Uint64() {
```

Comment:
You're correct that this is indeed wrong, however, data must be passed in as bytes and the front end the data should be passed as hex (hence the `*2`).

I propose that you properly fix this up by patching the console to validate for **hex** and that `SetExtra([]byte(...))` is done as `SetExtra(common.Hex2Bytes(...))`